emhttp/plugins/dynamix/include/NetworkInfo.php

```diff
@@ -24,19 +24,18 @@ function port($eth) {
   $sys = "/sys/class/net";
   if (substr($eth,0,4)=='wlan') return $eth;
   $x = preg_replace('/[^0-9]/','',$eth);
-  return file_exists("$sys/br{$x}") ? "br${x}" : (file_exists("$sys/bond{$x}") ? "bond{$x}" : "eth{$x}");
+  return file_exists("$sys/br{$x}") ? "br{$x}" : (file_exists("$sys/bond{$x}") ? "bond{$x}" : "eth{$x}");
 }
 
 exec("grep -Po 'nameserver \K\S+' /etc/resolv.conf 2>/dev/null",$ns);
-$eth    = $_POST['port'];
-$vlan   = $_POST['vlan'];
+$eth    = $_POST['port'] ?? '';
+$vlan   = $_POST['vlan'] ?? '';
+$wlan0  = $eth == 'wlan0';
 $port   = port($eth).($vlan ? ".$vlan" : "");
 $v6on   = trim(file_get_contents("/proc/sys/net/ipv6/conf/$port/disable_ipv6"))==='0';
 $none   = _('None');
 $error  = "<span class='red-text'>"._('Missing')."</span>";
 $note   = in_array($eth,['eth0','wlan0']) && !$vlan ? $error : $none;
-$link   = _(ucfirst(exec("ethtool $eth 2>/dev/null | awk '$1==\"Link\" {print $3;exit}'")) ?: 'Unknown')." ("._(exec("ethtool $eth 2>/dev/null | grep -Pom1 '^\s+Port: \K.*'") ?: ($eth=='wlan0' ? 'wifi' :'not present')).")";
-$speed  = _(preg_replace(['/^(\d+)/','/!/'],['$1 ',''],exec("ethtool $eth 2>/dev/null | awk '$1==\"Speed:\" {print $2;exit}'")) ?: 'Unknown');
 $ipv4   = array_filter(explode(' ',exec("ip -4 -br addr show $port scope global 2>/dev/null | awk '{\$1=\$2=\"\";print;exit}' | sed -r 's/ metric [0-9]+//g; s/\/[0-9]+//g'")));
```

Comment:
Let's make a habit of wrapping all variables in `exec` statements with `escapeshellarg`, so 
`exec("ip -4 -br addr show ".escapeshellarg($port)." scope global ...`

Reply:
I added the security layer